### DISCUSSION TITLE
Console errors when parsing JSON string (SSR)

### DISCUSSION THREAD

**Discussion by angelov-a** [urql-graphql/urql#1968] [Q&A] [answered]
Hello,

My app displays the data but there are console errors. I have tried several ways of parsing the string.

Do you have any suggestions?

Thanks in advance.

```javascript
JSON.parse(${urqlData})

// the passed value is an object
// Uncaught SyntaxError: Unexpected token o in JSON at position 1
```

```javascript
JSON.parse(\`${urqlData}\`)

// VM347:1 Uncaught SyntaxError: Unexpected token c in JSON at position 24
```

```javascript
JSON.parse('${urqlData}')

// doesn't work because there is a ' in one of the strings
// Uncaught SyntaxError: missing ) after argument list
```

```javascript
JSON.parse("${urqlData}")

// doesn't work because there are unescaped "
// Uncaught SyntaxError: missing ) after argument list
```

My configuration

```javascript
const urqlData = JSON.stringify(ssrCache.extractData());

const html = template
    .replace(`<!--preload-links-->`, preloadLinks)
    .replace(`<!--ssr-outlet-->`, appHtml)
    .replace(
        `/*urqlData*/`,
        `window.__URQL_DATA__ = JSON.parse(${urqlData});`
    );


// query
{
    characters {
        info {
            count
        }
        results {
            id
            name
            type
            status
            image
            location {
                name
                type
                dimension
            }
        }
    }
}

// url
https://rickandmortyapi.com/graphq
```

**Comment by kitten** [comment, marked as answer]
There are several solutions here:

- don't use `JSON.parse` and just leave it out (JSON is always a subset of JS)
- if you choose to use backticks i.e. template literals (and if you can accept that IE11 won't be supported), you not only need to escape the backticks but also the `$` sign, as in `\$`, since you're writing out an interpolation in what's already a template string
- if you'd like to use `JSON.parse` on your JSON string, then you need to turn your JSON string into a string, so ironically you'd either have to manually replace the quotes with escaped versions or.... call `JSON.stringify` again 😄 `JSON.parse(${JSON.stringify(str)})` will do it

**Reply by angelov-a** [reply]
Thanks, @kitten. That was helpful :)
